src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/ApplyBQSRDataflow.java

```diff
@@ -96,10 +107,43 @@ protected void setupPipeline(Pipeline pipeline) {
         final SAMSequenceDictionary sequenceDictionary = header.getSequenceDictionary();
         final List<SimpleInterval> intervals = intervalArgumentCollection.intervalsSpecified() ? intervalArgumentCollection.getIntervals(sequenceDictionary) :
                 IntervalUtils.getAllIntervalsForReference(sequenceDictionary);
-        final BaseRecalOutput recalInfo = new BaseRecalOutput(BQSR_RECAL_FILE);
-        PCollection<BaseRecalOutput> recalInfoSingletonCollection = pipeline.apply(Create.of(recalInfo).withName("recal_file ingest"));
+        PCollection<BaseRecalOutput> recalInfoSingletonCollection;
+        if (BucketUtils.isCloudStorageUrl(BQSR_RECAL_FILE_NAME)) {
+            recalInfoSingletonCollection = BaseRecalOutputSource.of(pipeline, BQSR_RECAL_FILE_NAME);
+        } else{
+            final BaseRecalOutput recalInfo = new BaseRecalOutput(new File(BQSR_RECAL_FILE_NAME));
+            recalInfoSingletonCollection = pipeline.apply(Create.of(recalInfo).withName("recal_file ingest"));
+        }
         PCollection<Read> output = readsSource.getReadPCollection(intervals, ValidationStringency.SILENT)
                 .apply(new ApplyBQSRTransform(header, recalInfoSingletonCollection, bqsrOpts));
-        SmallBamWriter.writeToFile(pipeline, output, header, OUTPUT);
+        intermediateGCSBam = OUTPUT;
+        if (isRemote() && !BucketUtils.isCloudStorageUrl(intermediateGCSBam)) {
```

Comment:
That wouldn't work because I modify intermediateGCSBam, so the function would return the wrong value the second time around.